### PR TITLE
chore: use the nested if's linter to reduce complexity globally

### PR DESCRIPTION
**Pull request by faddat**
## 1. Summary
Enable some new linters I learned while fixing CometBFT tests

* https://github.com/cometbft/cometbft/pull/2005

**notes**

* When main is merged into this PR, issues crop up fast.  Can someone try merging main here?  I haven't figured out what the issue is yet.
* we will wait until after the next release is out to merge this PR

## 2.Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 5. Checklist

- [ ] Does the Readme need to be updated?

## Summary by CodeRabbit

- **Refactor**
    - Improved type safety by changing types from `interface{}` to `any` in various components.
    - Refactored airdrop functionality for enhanced clarity and logic.
    - Organized claims manager code for better clarity.
    - Updated interchain functionality with improved logic and clarity.
    - Simplified method signatures in several components for better readability.
    - Enhanced message dispatching in `CustomMessenger` for `TokenFactoryMsg` with clearer logic.
    - Improved delegation logic in interchain staking with new functions and refined error handling.
- **Chores**
    - Updated linter configurations for better code quality checks.

### CODE REVIEW

File: x/interchainstaking/keeper/ibc_packet_handlers.go [hunk 1]
```diff
@@ -473,6 +473,16 @@ func (k *Keeper) HandleWithdrawForUser(ctx sdk.Context, zone *types.Zone, msg *b
...
+		return false
+	}
+	if msg.Amount[0].Denom != withdrawalRecord.Amount[0].Denom {
+		return false
+	}
+	return withdrawalRecord.Amount.IsEqual(msg.Amount)
+}
+
 func (k *Keeper) GCCompletedRedelegations(ctx sdk.Context) error {
 	var err error
```
[the remark this answers is not in the document]
Comment (faddat) [reply x/interchainstaking/keeper/ibc_packet_handlers.go:488]: I like this idea. Are you able to implement it?

**Comment by faddat** [comment]
Note: let's not merge this until a new release has been cut, due to the large logic refactors, this PR should run on a testnet.

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>48d8543</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1158.app-dev.quicksilver.zone'>https://pr-1158.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/GKvKCgdzEdW2vJGSLAcmcohgMuA9'>https://vercel.com/quicksilver/quicksilver/GKvKCgdzEdW2vJGSLAcmcohgMuA9</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8464124223)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>d873bc6</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1158.app-dev.quicksilver.zone'>https://pr-1158.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/F5DhQ7czZBtT24eBhV3n4wpdmH9y'>https://vercel.com/quicksilver/quicksilver/F5DhQ7czZBtT24eBhV3n4wpdmH9y</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8478112995)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>690bd5c</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1158.app-dev.quicksilver.zone'>https://pr-1158.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/ERZimtwTUWqm3Nn3op5sAQwGsiKb'>https://vercel.com/quicksilver/quicksilver/ERZimtwTUWqm3Nn3op5sAQwGsiKb</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8596270242)

**Comment by joe-bowman** [comment]
This pull request has been automatically marked as stale because it has not had any recent activity. It will be closed if no further activity occurs. Thank you!

**Comment by tropicaldog** [comment]
should enable this linter after #1202 done to avoid conflicts. should close this for now sir @faddat

**Comment by faddat** [comment]
Great idea.  I'm going to close it.

**Pull request closed without merging**